mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -696,23 +700,55 @@ impl NegotiationDataMsg {
     }
 }
 
-/// Data to be exchanged and validated on swap start, the replacement of LP_pubkeys_data, LP_choosei_data, etc.
-#[derive(Debug, Default, Deserializable, Eq, PartialEq, Serializable)]
-struct SwapNegotiationData {
-    started_at: u64,
-    payment_locktime: u64,
-    secret_hash: H160,
-    persistent_pubkey: H264,
+#[derive(Clone, Debug, Eq, Deserialize, PartialEq, Serialize)]
+pub struct PaymentDataV2 {
+    data: Vec<u8>,
+    // Next step instructions for the other side whether taker or maker.
+    // An example for this is a maker/taker sending the taker/maker a lightning invoice to be payed.
+    next_step_instructions: Vec<u8>,
+}
+
+#[derive(Clone, Debug, Eq, Deserialize, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum SwapTxDataMsg {
```

Comment:
Perhaps, V1 and V2 are not correct names, because V2 will be used for lightning swaps only. Am I right?
In this case, how about
```rust
SwapTxDataMsg {
    Regular(Vec<u8>),
    WithInstructions(PaymentWithInstructions),
}
```

Reply:
> Perhaps, V1 and V2 are not correct names, because V2 will be used for lightning swaps only. Am I right?

Yes, you are right, it is only used for lightning swaps for now. But instructions can be used in any future protocol integration that requires the maker/taker sending the other side instructions on how to make the payment.

> In this case, how about
> 
> SwapTxDataMsg {
>     Regular(Vec<u8>),
>     WithInstructions(PaymentWithInstructions),
> }

It looks more clear with these names :)